tests/run/C_argc_argv-to-java_args/src/main/scala/CArgcArgvToJavaArgs.scala

```diff
@@ -0,0 +1,10 @@
+object CArgcArgvToJavaArgs {
+  def main(args: Array[String]) {

```

Comment:
I think he meant
def main(args: Array[String]) {

should be
def main(args: Array[String]) = {

On Fri, Aug 5, 2016, 13:40 Francois Bertrand <notifications@github.com>
wrote:

> In
> tests/run/C_argc_argv-to-java_args/src/main/scala/CArgcArgvToJavaArgs.scala
> <https://github.com/scala-native/scala-native/pull/259#discussion_r73631053>
> :
>
> > @@ -0,0 +1,10 @@
> > +object CArgcArgvToJavaArgs {
> > +  def main(args: Array[String]) {
>
> Procedural or imperative style is always hard to beat if you have concerns
> about code size, memory consumption and eficiency.
>
> For example, code like for (i <- 0 until� len) create objects under the
> scene.
>
> FInally, the following is a educated guess, I would have to try it to be
> sure. The LLVM backends are optimized for imperative language like C/C++.
> Functional style will generate IR code that will be difficult to optimize
> by these backend compilers. As more functional language target LLVM, more
> functional specific optimization pass will be added.
>
> Until then, I believe it's better to stick to imperative style in a
> heavily used library.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/scala-native/scala-native/pull/259/files/ea9c87de71ea11c14fdb543876875bb5a0f25835#r73631053>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AEG1Z4_8Gn2JOude1W5xP9tnghb6O0Oyks5qcpScgaJpZM4Jb-SL>
> .
>